src/lockup.rs

```diff
@@ -39,15 +39,20 @@ impl Lockup {
         }
     }
 
-    pub fn claim(&mut self, index: LockupIndex) -> LockupClaim {
+    pub fn claim(&mut self, index: LockupIndex, claim_amount: Balance) -> LockupClaim {
         let unlocked_balance = self.schedule.unlocked_balance(current_timestamp_sec());
-        assert!(unlocked_balance >= self.claimed_balance, "Invariant");
-        let unclaimed_balance = unlocked_balance - self.claimed_balance;
-        self.claimed_balance = unlocked_balance;
+        let balance_claimed_new = self.claimed_balance + claim_amount;
```

Comment:
This may overflow since it relies on user's input

Reply:
Fixed.
By the way, what is the best practice for handling overflows?
Can't I just rely on `[profile.release] overflow-checks = true` ?

Reply:
It's fine, but ideally they are explicitly handled. Either using assert/require macro or using `checked_sub` and `checked_add`.